io-engine/src/bdev/nexus/mod.rs

```diff
@@ -173,15 +174,19 @@ pub async fn shutdown_nexuses() {
 
     for mut nexus in nexuses.into_iter() {
         // Destroy nexus and persist its state in the ETCd.
-        if let Err(error) = nexus.as_mut().destroy_ext(true).await {
-            error!(
-                name = nexus.name,
-                error = error.verbose(),
-                "Failed to destroy nexus"
-            );
+        match nexus.as_mut().destroy_ext(true).await {
+            Ok(_) => {
+                nexus.event(EventAction::Shutdown).generate();
+            }
+            Err(error) => {
+                error!(
```

Comment:
Wonder if maybe shutdown can include a result?

Reply:
What would the result contain here?

Reply:
Added the error to event details.